security/pkg/k8s/controller/monitoring.go

```diff
@@ -15,78 +15,72 @@
 package controller
 
 import (
-	"github.com/prometheus/client_golang/prometheus"
+	"istio.io/pkg/monitoring"
 )
 
 const (
 	errorlabel = "error"
 )
 
 var (
-	serviceAccountCreationCounts = prometheus.NewCounterVec(prometheus.CounterOpts{
-		Namespace: "citadel",
-		Subsystem: "secret_controller",
-		Name:      "svc_acc_created_cert_count",
-		Help:      "The number of certificates created due to service account creation.",
-	}, []string{})
+	errorTag = monitoring.MustCreateLabel(errorlabel)
 
-	serviceAccountDeletionCounts = prometheus.NewCounterVec(prometheus.CounterOpts{
-		Namespace: "citadel",
-		Subsystem: "secret_controller",
-		Name:      "svc_acc_deleted_cert_count",
-		Help:      "The number of certificates deleted due to service account deletion.",
-	}, []string{})
+	serviceAccountCreationCounts = monitoring.NewSum(
+		"svc_acc_created_cert_count",
+		"The number of certificates created due to service account creation.",
+	)
 
-	secretDeletionCounts = prometheus.NewCounterVec(prometheus.CounterOpts{
-		Namespace: "citadel",
-		Subsystem: "secret_controller",
-		Name:      "secret_deleted_cert_count",
-		Help:      "The number of certificates recreated due to secret deletion (service account still exists).",
-	}, []string{})
+	serviceAccountDeletionCounts = monitoring.NewSum(
+		"svc_acc_deleted_cert_count",
+		"The number of certificates deleted due to service account deletion.",
+	)
 
-	csrErrorCounts = prometheus.NewCounterVec(prometheus.CounterOpts{
-		Namespace: "citadel",
-		Subsystem: "secret_controller",
-		Name:      "csr_err_count",
-		Help:      "The number of errors occurred when creating the CSR.",
-	}, []string{})
+	secretDeletionCounts = monitoring.NewSum(
+		"secret_deleted_cert_count",
+		"The number of certificates recreated due to secret deletion (service account still exists).",
+	)
 
-	certSignErrorCounts = prometheus.NewCounterVec(prometheus.CounterOpts{
-		Namespace: "citadel",
-		Subsystem: "secret_controller",
-		Name:      "csr_sign_err_count",
-		Help:      "The number of errors occurred when signing the CSR.",
-	}, []string{errorlabel})
+	csrErrorCounts = monitoring.NewSum(
+		"csr_err_count",
+		"The number of errors occurred when creating the CSR.",
+	)
+
+	certSignErrorCounts = monitoring.NewSum(
```

Comment:
I tested it and it looks like you need `monitoring.WithLabels(errorTag)` here as well, before using it with tag values.

Reply:
Thanks for the catch @pitlv2109, should be fixed now